satpy/readers/nwcsaf_nc.py

```diff
@@ -414,3 +407,16 @@ def remove_empties(variable):
             variable.attrs.pop(key)
 
     return variable
+
+
+def read_nwcsaf_time(time_value):
+    """Read the time, nwcsaf-style."""
+    try:
+        # MSG:
+        try:
+            return datetime.strptime(time_value, '%Y-%m-%dT%H:%M:%SZ')
+        except TypeError:
```

Comment:
Is that possibly for older NWCSAF files? Would be good to have it tested in that case.

Reply:
Ok, so looking at this, I think both recent version of netcdf and h5netcdf do the conversion to string automatically, so this part can be removed in the future I think. For now, I can't reproduce a netcdf that requires this, so I will leave this untested unfortunately.